src/main/java/gregtech/common/items/behaviors/spray/CreativeSprayBehavior.java

```diff
@@ -0,0 +1,341 @@
+        }
+
+        return I18n.format(unlocalizedName, colorString);
+    }
+
+    @SideOnly(Side.CLIENT)
+    @Override
+    public void handleMouseEventClient(@NotNull MouseEvent event, @NotNull EntityPlayerSP playerClient,
+                                       @NotNull EnumHand hand, @NotNull ItemStack sprayCan) {
+        // Middle click pressed down
+        if (event.getButton() == 2 && event.isButtonstate()) {
+            event.setCanceled(true);
+
+            RayTraceResult rayTrace = RayTracer.retrace(playerClient);
+            if (rayTrace != null && rayTrace.typeOfHit == RayTraceResult.Type.BLOCK) {
+                World world = playerClient.world;
+                BlockPos pos = rayTrace.getBlockPos();
+                EnumFacing facing = rayTrace.sideHit;
+                ColoredBlockContainer container = ColoredBlockContainer.getContainer(world, pos, facing,
+                        playerClient);
+
+                if (container.isValid(world, pos, facing, playerClient)) {
+                    if (usesRGB(sprayCan) && container.supportsARGB() &&
+                            !container.colorMatches(world, pos, facing, playerClient, getColorInt(sprayCan))) {
+                        int color = container.getColorInt(world, pos, facing, playerClient);
+                        if (color != -1) {
+                            setColor(sprayCan, color);
+                            sendToServer(hand, buf -> buf
+                                    .writeByte(1)
+                                    .writeInt(color));
+                            return;
+                        }
+                    } else if (!container.colorMatches(world, pos, facing, playerClient, getColor(sprayCan))) {
+                        EnumDyeColor color = container.getColor(world, pos, facing, playerClient);
+                        if (color != null) {
+                            setColor(sprayCan, color);
+                            sendToServer(hand, buf -> buf
+                                    .writeByte(2)
+                                    .writeByte(color.ordinal()));
+                            return;
+                        }
+                    }
+                }
+            }
+
+            // If the player isn't sneaking and wasn't looking at a colored block, open gui
+            sendToServer(hand, buf -> buf.writeByte(0));
+        }
+    }
+
+    @Override
+    public void handleMouseEventServer(@NotNull PacketBuffer buf, @NotNull EntityPlayerMP playerServer,
+                                       @NotNull EnumHand hand, @NotNull ItemStack sprayCan) {
+        switch (buf.readByte()) {
+            case 0 -> MetaItemGuiFactory.open(playerServer, EnumHand.MAIN_HAND);
+            case 1 -> setColor(sprayCan, buf.readInt());
+            case 2 -> setColor(sprayCan, EnumDyeColor.values()[buf.readByte()]);
+        }
+    }
+
+    private static class InterceptedPageController extends PagedWidget.Controller {
```

Comment:
Do we need these overrides here? What does the crafting station do, it also has paged/tabbed widgets, and I don't think it does all this.